integration_v3/test_backup_v4.py

```diff
@@ -484,3 +485,43 @@ def test_cancel_backup(client: weaviate.WeaviateClient) -> None:
         if status_resp.status == BackupStatus.CANCELED:
             break
         time.sleep(0.1)
+    status_resp = client.backup.get_create_status(backup_id=backup_id, backend=BACKEND)
+    assert status_resp.status == BackupStatus.CANCELED
+        
+def test_cancel_backup_with_waiting(client: weaviate.WeaviateClient) -> None:
+    """Cancel a backup while waiting for completion."""
+    backup_id = _create_backup_id()
+    if client._connection._weaviate_version.is_lower_than(1, 24, 25):
+        pytest.skip("Cancel backups is only supported from 1.24.25")
+
+    # Run the backup creation in a separate thread
+
+    def create_backup():
+        nonlocal resp
+        with pytest.raises(BackupFailedException) as excinfo:
+            client.backup.create(
+                backup_id=backup_id,
+                backend=BACKEND,
+                wait_for_completion=True,
+            )
+        assert "Backup was canceled" in str(excinfo.value)
+
+    resp = None
+    backup_thread = threading.Thread(target=create_backup)
+    backup_thread.start()
+
+    # Cancel the backup
+    assert client.backup.cancel_backup(backup_id=backup_id, backend=BACKEND)
+
+    # Wait for the backup thread to finish
+    backup_thread.join()
```

Comment:
I think this test will be very flaky as it requires exact timing to work - could you rewrite it so it won't fail in case the cancel does not happen at the "correct" time?

Reply:
Looking into this for the TS client, I'm finding it particularly difficult to actually get the `CANCELED` state returned as I think it has to be timed perfectly between `STARTED` and `SUCCESS`, which is a very small window when the collections themselves are small

IMO: This functionality should be tested within the client repos using mocks as a result with the full integration test living in the `weaviate-e2e-tests` repo. wdyt @jfrancoa?

Reply:
yeah, a mock test would be good - I dont think that we need a big e2e test. This is purely client behaviour

Reply:
I added the mock test!